runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java

```diff
@@ -257,6 +262,19 @@ public void onGameObjectSpawned(GameObjectSpawned event)
 			}
 		}
 	}
+	
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		// Check for the Fire option being selected on the cannon.
+		if (event.getMenuAction() != MenuAction.GAME_OBJECT_FIRST_OPTION || event.getId() != ObjectID.DWARF_MULTICANNON)
```

Comment:
This should also support using cannonballs on a cannon as that would also reload a cannon.

Reply:
Good shout. Fixed, and also found (and fixed) a weird tick-perfect issue where the indicator could appear on another cannon.